test/test_selections.py

```diff
@@ -87,37 +88,44 @@ def container_on_disk_distributed():
 def test_H5FileSelect(container_on_disk):
     """Tests that makes hdf5 objects and tests selecting on their axes."""
 
-    m = SiderealGainData.from_file(container_on_disk, freq_sel=fsel, input_sel=isel)
-    assert np.all(m["gain"][:] == dset1[(fsel, isel, slice(None))])
-    assert np.all(m["weight"][:] == dset2[(fsel, slice(None))])
+    m = GainData.from_file(
+        container_on_disk, freq_sel=fsel, input_sel=isel, time_sel=tsel
+    )
+    assert np.all(m["gain"][:] == dset1[(fsel, isel, tsel)])
+    assert np.all(m["weight"][:] == dset2[(fsel, tsel)])
     assert np.all(m.index_map["freq"] == freqs[fsel])
     assert np.all(m.index_map["input"] == inputs[isel])
+    assert np.all(m.index_map["time"] == inputs[tsel])
```

Comment:
```suggestion
    assert np.all(m.index_map["time"] == times[tsel])
```

Reply:
Ooops. This is a fail! I'll post a new commit to fix all this up.